GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java

```diff
@@ -348,6 +364,65 @@ public InstanceData parseInstance(Sheet sheet, int startRow,
                             }
                             break;
 
+                        case OPTION:
+                            // Two different possible formats:
+                            // With codes: code1:val1|code2:val2|...
+                            // Without codes: val1|val2|...
+                            String optionString = ExportImportUtils.parseCellAsString(cell);
+                            if (optionString.isEmpty()) {
+                                break;
+                            }
+                            String[] optionParts = optionString.split("\\|");
+                            List<Map<String, Object>> optionList = new ArrayList<>();
+                            for (String optionNode : optionParts) {
+                                String[] codeAndText = optionNode.split(":");
+                                Map<String, Object> optionMap = new HashMap<>();
+                                if (codeAndText.length == 1) {
+                                    optionMap.put("text", codeAndText[0]);
+
+                                } else if (codeAndText.length == 2) {
+                                    optionMap.put("code", codeAndText[0]);
+                                    optionMap.put("text", codeAndText[1]);
+                                } else {
+                                    log.warn("Invalid option node: " + optionNode);
+                                }
+                                optionList.add(optionMap);
+                            }
+
+                            // Should we add the 'allowOther' flag to the last node?
+                            if (Boolean.TRUE.equals(questionDto.getAllowOtherFlag())
+                                    && !optionList.isEmpty()) {
+                                Map<String, Object> lastNode = optionList
+                                        .get(optionList.size() - 1);
+                                String lastNodeText = (String) lastNode.get("text");
+                                boolean isOther = true;
+                                List<QuestionOptionDto> existingOptions = optionNodes
+                                        .get(questionId);
+                                if (existingOptions != null && lastNodeText != null) {
+                                    // Only compare text. Is that enough?
```

Comment:
I believe comparing text is enough. We probably don't want a situation where two options have the same text and different codes since the app user sees the text.